src/convertToObject.js

```diff
@@ -6,7 +6,69 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const rawLines = sourceString.split(';');
+
+  // filter lines without letters
+  const noEmptyLines = rawLines
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
+
+  const aggregatedLines = [];
+
+  for (const line of noEmptyLines) {
+    if (line.includes(':')) {
+      aggregatedLines.push(line);
+    } else {
+      aggregatedLines[aggregatedLines.length - 1] += line;
+    }
+  }
+
+  let separatedRules = [];
+
+  for (const line of aggregatedLines) {
+    if (line.split(':').length >= 3) {
+      separatedRules = [...separatedRules, ...line.split(';')];
+    } else {
+      separatedRules.push(line);
+    }
+  }
+
+  separatedRules = separatedRules.filter((item) => item.length > 1);
```

Comment:
[CODE KNOWLEDGE] - Don't need to make a copy before each operation. Moreover, methods like `filter` don't mutate the original array. You can chain this `filter` directly to the previous operation where `separatedRules` is defined.